src/components/Viewport.ts

```diff
@@ -1476,4 +1675,161 @@ export default class Viewport {
       } as Partial<NeedPanelEvent>,
     );
   }
+
+  private updateVisiblePanels() {
+    const state = this.state;
+    const options = this.options;
+    const isHorizontal = this.options.horizontal;
+    const cameraElement = this.cameraElement;
+    const visibleIndex = state.visibleIndex;
+
+    if (!options.renderOnlyVisible) {
+      return;
+    }
+
+    if (!this.nearestPanel) {
+      this.resetVisibleIndex();
+      while (cameraElement.firstChild) {
+        cameraElement.removeChild(cameraElement.firstChild);
+      }
+      return;
+    }
+
+    const panelManager = this.panelManager;
+    const allPanelCount = panelManager.getRange().max + 1;
+    const cloneCount = panelManager.getCloneCount();
+    const cameraPos = this.getCameraPosition();
+    const basePanel = this.nearestPanel;
+    const viewportBbox = this.getBbox();
+    const bbox = this.options.overflow
+      ? this.getWindowBBox()
+      : viewportBbox;
+    const viewportPos = isHorizontal
+      ? viewportBbox.x
+      : viewportBbox.y;
+
+    const checkLastPanel = (
+      panel: Panel,
+      getNextPanel: (panel: Panel) => Panel | null,
+      isOutOfViewport: (panel: Panel) => boolean,
+    ): Panel => {
+      let lastPanel = panel;
+      while (true) {
+        const nextPanel = getNextPanel(lastPanel);
+        if (!nextPanel || isOutOfViewport(nextPanel)) {
+          break;
+        }
+        lastPanel = nextPanel;
+      }
+      return lastPanel;
+    };
+
+    const lastPanelOfNextDir = checkLastPanel(basePanel, panel => {
+      const nextPanel = panel.nextSibling;
+
+      if (!nextPanel || nextPanel.getPosition() < panel.getPosition()) {
```

Comment:
Do you have reason to use negative comparison? like performance...

If you don't, how about to use positive comparison?
```ts
if (nextPanel && nextPanel.getPosition() >= panel.getPosition()) {
  return nextPanel;
} else {
  return null;
}
```

Reply:
I think I don't have a reason for that. It's just what I've used to.
Can you give me some suggestion why the positive comparison is better?

Reply:
I think it's more readable to make condition by what we want to get. In above code, we want to get nextPanel (if exist) so I think it's proper to use condition for getting a `nextPanel`.

Reply:
That's quite convincing, I'll fix that.